### PR TITLE
Add a note explaining to wait for smooch.init before updating user data

### PR DESCRIPTION
**Pull request by Go-Fred**
My pull request also includes a change in rest... I don't know why...

### CODE REVIEW

File: source/includes/users/_user_data_note_js.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+<aside class="notice">
+You need to make sure the SDK is properly initialized before attempting to update a user information.
+</aside>
+
+You can either update the user right after the initialization using ".then":
+
+```javascript
+<script>
```
Comment (Mario54) [inline source/includes/users/_user_data_note_js.md]: Indentation is off in this code sample I think

File: source/rest/index.md [hunk 2]
```diff
@@ -1057,7 +1057,14 @@ curl https://api.smooch.io/v1/appusers/c7f6e6d6c3a637261bd9656f/conversation/ima
      -F 'name=Steve'
 ```
 ```js
-// Not supported yet in the JS SDK
+smooch.conversations.uploadImage('c7f6e6d6c3a637261bd9656f', source: './screenshot.jpg',
```
Comment (Mario54) [inline source/rest/index.md]: Hmm I think the call is different on the server and on the browser, right @lemieux?

Comment (lemieux) [inline source/rest/index.md]: ehhh this is not right at all in fact.

Comment (Go-Fred) [inline source/rest/index.md]: What are the arguments then? It's not this: https://github.com/smooch/smooch-core-js/blob/integration/src/api/conversations.js#L61 ?

Comment (lemieux) [inline source/rest/index.md]: the `source` parameter must be a `Blob` instance on the frontend or a Readable stream in the backend. And it doesn't work in the backend yet.

Comment (Go-Fred) [inline source/rest/index.md]: Is there a way to write it in a simple way for this section?

File: source/includes/users/_user_data_note_js.md (hunk 1 above)
Comment (Go-Fred) [inline source/includes/users/_user_data_note_js.md]: Is it because of the aside?

File: source/rest/index.md (hunk 2 above)
Comment (lemieux) [inline source/rest/index.md]:
```
// Frontend version
var file = fileInput.files[0];
smooch.conversations.uploadImage('c7f6e6d6c3a637261bd9656f', file, { ... });

// Not yet supported on Node.
```

Comment (Go-Fred) [inline source/rest/index.md]: Thanks! And for the rest is ok? 
{ text: 'Just put some vinegar on it',
  role: 'appUser'
})

Or should I leave it like you said {...}

Comment (Go-Fred) [inline source/rest/index.md]: Also is it working for both AppUser & AppMaker?

Comment (lemieux) [inline source/rest/index.md]: keep what you had at the end, and it should be working for both

File: source/includes/users/_user_data_note_js.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+Smooch.init({
+  appToken: 'your-app-token',
+  emailCaptureEnabled: true,
```
Comment (lemieux) [inline source/includes/users/_user_data_note_js.md]: removing the dangling comma

**Comment by Go-Fred** [comment]
@Mario54 @lemieux Done! Thanks for the comments!

**Comment by Mario54** [comment]
👍

**Pull request merged**